### PR TITLE
IgxOverlay topics refactor

### PR DESCRIPTION
**Pull request by ViktorSlavov**
Refactor articles (headers), add code snippets, add more sample anchors

### CODE REVIEW

File: en/components/overlay_main.md [hunk 1]
```diff
@@ -4,25 +4,164 @@ _description: Provides a service which enables developers to position content ab
...
 </p>
 <div class="divider--half"></div>
 
-## Overlay Demo
-<p class="highlight">
-The below demo illustrates how the overlay service `overlaySettings` can be manipulated in order to alter the way in which the overlay element is shown.
-</p>
-<div class="sample-container loading" style="height: 600px">
-    <iframe id="overlay-sample-iframe" frameborder="0" seamless width="100%" height="100%" src="{environment:demosBaseUrl}/overlay-sample" onload="onSampleIframeContentLoaded(this);"></iframe>
+## Usage
```
Comment (simeonoff) [inline en/components/overlay_main.md]: This is not needed.

File: en/components/overlay_main.md [hunk 2]
```diff
@@ -4,25 +4,164 @@ _description: Provides a service which enables developers to position content ab
...
+<button (click)="showInOverlay()">Show Overlay</button>
+</div>
+
+```
+<div class="divider--half"></div>
+
+The overlay service `show()` method accepts 2 arguments, the first one being the content that should be rendered in the overlay. There are a couple of different scenarios how the content can be passes:
+  - A component definition (illustrated in the sample above) - When passing a component in as the first argument, the overlay service creates a new instance of that component and dynamically attaches it to the `overlay` DOM.
+  - An existing piece of DOM - Any view that is already rendered on the page can be passed through the overlay service and be rendered in the overlay DOM. Using this approach will:
+    - Get the reference to the passed view from the Angular
```
Comment (simeonoff) [inline en/components/overlay_main.md]: Should be `Get the reference to the passed view from Angular`

File: en/components/overlay_main.md [hunk 3]
```diff
@@ -4,25 +4,164 @@ _description: Provides a service which enables developers to position content ab
...
+    @ViewChild(`myAnchorButton`)
+    private myAnchorButton: ElementRef;
+
+    public toggleOverlay() {
+        if (!this._overlayShown) { // If the element is not visible, show it
+            this._overlayId = this.overlayService.show(MyDynamicComponent, {
+                positionStrategy: new ConnectedPositioningStrategy({ target: this.myAnchorButton })
+            }); // The show method returns an ID that can be used to reference the shown content
+        } else { // If the element is not visible, hide it
+            this.overlayService.hind(this._overlayId); // Find and remove the component from the overlay container
```
Comment (simeonoff) [inline en/components/overlay_main.md]: there is a typo - `hind` should be `hide`

File: en/components/overlay_main.md [hunk 4]
```diff
@@ -4,25 +4,164 @@ _description: Provides a service which enables developers to position content ab
...
+<!-- in my-overlay-component.component.html -->
+<div class='content'>
+...
+<button (click)="showInOverlay()">Show Overlay</button>
+</div>
+
+```
+<div class="divider--half"></div>
+
+The overlay service `show()` method accepts 2 arguments, the first one being the content that should be rendered in the overlay. There are a couple of different scenarios how the content can be passes:
```
Comment (bazal4o) [inline en/components/overlay_main.md]: Should be content can be passed

File: en/components/overlay_main.md [hunk 5]
```diff
@@ -30,7 +28,7 @@ export class MyOverlayComponent {
 ...
 ```
 
-### Displaying overlay content
+## Displaying overlay content
```
Comment (simeonoff) [inline en/components/overlay_main.md]: This should be an H3 headline, not H2. The same goes for every other change made below.

**Pull request merged**